javascript/javascript2/week3/warmUp.js

```diff
@@ -0,0 +1,192 @@
+
+setTimeout(afterSeconds, 2500)
+
+function afterSeconds() {
+    console.log('Called after 2.5 seconds')
+}
+
+// 2. Create a function that takes 2 parameters: delay and stringToLog. Calling this function should log out the
+//    stringToLog after delay seconds. Call the function you have created with some different arguments.
+
+function logOutWithDelay(delay, stringToLog) {
+    function stringToLogFunction() {
+        console.log(stringToLog)
+    }
+    delayWithSeconds = delay * 1000
+    setTimeout(stringToLogFunction, delayWithSeconds)
+}
+
+logOutWithDelay(3, 'This string logged after 3 seconds')
+logOutWithDelay(5, 'This string logged after 5 seconds')
+
+// 3. Create a button in html. When clicking this button, use the function you created in the previous task to log out the text:
+//    Called after 5 seconds 5 seconds after the button is clicked.
+
+const button = document.getElementById('clickButton')
+
+button.addEventListener('click', () => {
+    logOutWithDelay(5, 'This string logged after 5 seconds')
+})
+
+
+// 4. Create two functions and assign them to two different variables. One function logs out Earth,
+//    the other function logs out Saturn. Now create a new third function that has one parameter: planetLogFunction.
+//    The only thing the third function should do is call the provided parameter function. Try call the third function
+//    once with the Earth function and once with the Saturn function.
+
+function earthLogger() {
+    console.log('Earth')
+}
+
+function saturnLogger() {
+    console.log('Saturn')
+}
+
+function planetLogFunction(functionForRun) {
+    const result = functionForRun()
+    return result
+}
+
+planetLogFunction(earthLogger)
+planetLogFunction(saturnLogger)
+
+// 5. Create a button with the text called "Log location". When this button is clicked the location
+//    (latitude, longitude) of the user should be logged out using this browser api
+// 6. Optional Now show that location on a map using fx the Google maps api
+
+const locationButton = document.getElementById('locationButton')
+const status = document.querySelector('#status')
+const logs = document.querySelector('#linkAndLogs')
+
+function getLatitudeLongitude() {
+    logs.href = '';
+    logs.textContent = '';
+
+    function confirmed(location) {
+        const latitude = location.coords.latitude;
+        const longitude = location.coords.longitude
+
+        status.textContent = '';
+        logs.href = `https://www.openstreetmap.org/#map=18/${latitude}/${longitude}`;
+        logs.textContent = `Latitude: ${latitude} °, Longitude: ${longitude} °`;
+    }
+
+    function error() {
+        status.textContent = 'Unable to retrieve your location'
+    }
+
+    if (!navigator.geolocation) {
+        status.textContent = 'Geolocation is not supported by your browser';
+    }
+    else {
+        status.textContent = 'Locating…';
+        navigator.geolocation.getCurrentPosition(confirmed, error);
+    }
+}
```

Comment:
This looks exactly like Simon's solution. Be sure to make it yourself, otherwise you will not learn it :)